src/platform_impl/macos/app_delegate.rs

```diff
@@ -106,6 +113,13 @@ extern "C" fn did_finish_launching(this: &Object, _: Sel, _: id) {
   trace!("Completed `applicationDidFinishLaunching`");
 }
 
+extern "C" fn application_should_terminate(_: &Object, _: Sel, _: id) -> c_int {
+  trace!("Triggered `applicationShouldTerminate`");
+  AppState::should_exit();
+  trace!("Completed `applicationShouldTerminate`");
+  NS_TERMINATE_LATER
```

Comment:
If I understand correctly, this will cause the application to never quit unless the event loop is changed. 

I see that in the associated tauri PR, you call `replyToApplicationShouldTerminate` which is fine I guess but the default behavior here should return `NS_TERMINATE_NOW` and need a way for that to be overridden.

Reply:
I didn't think of a good way to handle this problem.

Is it possible to transplant the processing method in tauri to tao? By default, `NSTerminateLater` is returned. If the user does not handle the event, reply to exit. Otherwise, reply, terminate or cancel according to the user's processing.

Reply:
there is a similar mechanism used when ScaleFactorChanged event is fired.